tensorflow/compiler/mlir/tools/kernel_gen/tf_to_gpu_binary.cc

```diff
@@ -64,8 +64,13 @@ int main(int argc, char** argv) {
     return 1;
   }
 
+#if TENSORFLOW_USE_ROCM
```

Comment:
Since both methods share the same first few lowering passes and have the same signature, it might make sense to have a common function to be called and do the specialization for Hsaco code later?

Reply:
Yes. we probably should have a single `GenerateGpuBinaryForTfCode` routine / call here, and the implementation of that routine, can have some ROCm / CUDA specific aspects to it. I will make that change in the next iteration. For this iteration, one of the goals was to leave the CUDA path as untouched as possible, to help make the rebasing easier.